templates/ts-apollo-fullstack/server/src/schema/schema.graphql

```diff
@@ -10,20 +10,27 @@ type Comment {
   note: Note
 }
 
-input CommentInput {
+input CommentData {
+  id: ID
+  text: String
+  description: String
+  noteId: ID
+}
+
+input CommentFilter {
   id: ID
   text: String
   description: String
   noteId: ID
 }
 
 type Mutation {
-  createNote(input: NoteInput): Note!
-  updateNote(input: NoteInput): Note!
-  deleteNote(input: NoteInput): Note!
-  createComment(input: CommentInput): Comment!
-  updateComment(input: CommentInput): Comment!
-  deleteComment(input: CommentInput): Comment!
+  createNote(data: NoteData!): Note!
+  updateNote(data: NoteData!): Note!
+  deleteNote(filter: NoteFilter!): Note!
+  createComment(data: CommentData!): Comment!
+  updateComment(data: CommentData!): Comment!
+  deleteComment(filter: CommentFilter!): Comment!
```

Comment:
@craicoverflow Missing rename

Reply:
`filter` is valid here, no? It should not be `data` since data is used to insert the data, whereas with a delete we are using the data to do a WHERE clause.

Maybe `filter` is misleading in all cases, `where` might be better?

Reply:
Filter is very confusing. For maintaining consistency we need delete to have more than just id (once again - offix cases) and we do not know what field is being used. Filter is very confusing as it suggest you can filter deltes (where it is all id based)

Reply:
This is where we should probably do an investigation and see how other systems do it.

Reply:
I will suggest to rename it to data and we can see later how to make it work